packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -75,10 +71,16 @@ export class DataSourceWithBackend<
     }
 
     const req: Promise<DataQueryResponse> = getBackendSrv()
-      .post('/api/ds/query', body)
+      .datasourceRequest({
+        url: '/api/ds/query',
+        method: 'POST',
+        data: body,
+        requestId,
+      })
       .then((rsp: any) => {
-        return this.toDataQueryResponse(rsp);
+        return this.toDataQueryResponse(rsp?.data);
```

Comment:
Why is the .data needed?

Reply:
Because the `datasourceRequest` returns a different response object than when using `post`. It returns https://github.com/grafana/grafana/blob/2fac83441364902908ce18980415f3029c603110/public/app/types/events.ts#L45-L55

Reply:
👍